cdk/stack/ynr.py

```diff
@@ -199,14 +252,93 @@ def create_cloudfront(
             "YNRCloudFront",
             default_behavior=cloudfront.BehaviorOptions(
                 origin=app_origin,
+                viewer_protocol_policy=cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
                 allowed_methods=cloudfront.AllowedMethods.ALLOW_ALL,
                 cache_policy=cloudfront.CachePolicy.CACHING_DISABLED,
+                compress=True,
             ),
             certificate=cert,
             domain_names=[fqdn],
             price_class=cloudfront.PriceClass.PRICE_CLASS_100,
         )
 
+        # Standard cache polity that allows CSRF headers and
+        # doesn't do much else. Caps cache at 5 minutes, to ensure
+        # we don't end up caching things for too long.
+        no_cache_not_authenticated = cloudfront.CachePolicy(
```

Comment:
Obviously naming things is hard, but this seems particularly badly named since we only use it for /admin (so the user always **is** authenticated) and it allows pages to be cached for up to 5 mins.
I'd actually be in favour of setting `max_ttl=Duration.minutes(0)` on this one as cacheing the admin is almost always not sensible. That said, django admin sends `cache-control: max-age=0, no-cache, no-store, must-revalidate, private` on every page so `max_ttl=Duration.minutes(5)` isn't really doing any damage.
Given that django sends `cache-control: max-age=0, no-cache, no-store, must-revalidate, private` on every page is this cache policy actually doing much extra over the default? I think on our other apps we just use the default policy and rely on django setting the right headers.

Reply:
Setting `max_ttl=Duration.minutes(0)` actually isn't possible here: doing so ends up disabling the cache and that means we can't set `header_behavior`. We want to be able to forward the headers in the list, and we need a policy that allows us to do this. Setting min and default to 0 means that nothing is actually cached, and the 5 minutes max just allows us to have a policy of some sort apply. 

I can try to change this, but I'm fairly sure that `cache_policy=cloudfront.CachePolicy.CACHING_DISABLED` prevents headers from being passed.

Reply:
Yep, confirmed that `CACHING_DISABLED` doesn't forward the CSRF headers. We need a policy with a non-0 default in order to forwards the headers.

Reply:
`forward_headers` is definitely more useful than `not_authenticated` :+1: 

I still feel like `no_cache` is odd given the policy can cache for up to 5 mins. How about `short_ttl_forward_headers` ?

Naming things is hard. Picking _something_ is probably more useful than bikeshedding this too much more.